pysr/regressor_sequence.py

```diff
@@ -0,0 +1,425 @@
+            X=historical_X,
+            y=current_X,
+            weights=weights,
+            variable_names=variable_names,
+            X_units=X_units,
+            y_units=y_units,
+            complexity_of_variables=complexity_of_variables,
+        )
+        return self
+
+    def predict(self, X, index=None, extra_predictions=0):
+        """
+        Predict y from input X using the equation chosen by `model_selection`.
+
+        You may see what equation is used by printing this object. X should
+        have the same columns as the training data.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Data of shape `(n_times, n_features)`.
+        index : int | list[int]
+            If you want to compute the output of an expression using a
+            particular row of `self.equations_`, you may specify the index here.
+            For multiple output equations, you must pass a list of indices
+            in the same order.
+        extra_predictions : int
+            If you want to predict more than one step into the future, specify
+            how many extra predictions you want. For example, if `extra_predictions=2`,
+            the model will predict the next two time points after the last time point
+            in `X`.
+
+        Returns
+        -------
+        x_predicted : ndarray of shape (n_samples, n_features)
+            Values predicted by substituting `X` into the fitted sequence symbolic
+            regression model.
+
+        Raises
+        ------
+        ValueError
+            Raises if the `best_equation` cannot be evaluated.
+        """
+        X = self._validate_data(X)
+        _check_assertions(X, recursive_history_length=self.recursive_history_length)
+        historical_X = np.lib.stride_tricks.sliding_window_view(
+            X.flatten(), self.recursive_history_length * np.prod(X.shape[1])
+        )[:: X.shape[1], :]
+        pred = self._regressor.predict(X=historical_X, index=index)
+        if extra_predictions > 0:
+            output = pred
+            previous_points = historical_X[-1]
+            # Without this, the model will re-predict the last data point
+            pred_once = self._regressor.predict(X=[previous_points], index=index)
+            previous_points = previous_points[X.shape[1] :]
+            previous_points = np.append(previous_points, pred_once)
+            previous_points = previous_points.flatten()
+            for _ in range(extra_predictions):
+                pred_once = self._regressor.predict(X=[previous_points], index=index)
+                previous_points = previous_points[X.shape[1] :]
+                previous_points = np.append(previous_points, pred_once)
+                previous_points = previous_points.flatten()
+                output = np.append(output, pred_once)
+            return output.reshape(-1, X.shape[1])
+        return pred
+
+    @classmethod
+    def from_file(
```

Comment:
I think it would be much simpler to just call `PySRRegressor.from_file`, and use that to instantiate `_regressor` in a new `PySRSequenceRegressor`